.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,258 @@
+
+  build-amd64:
+    runs-on: ubuntu-latest
+    container: 
+      image: rustlang/rust:nightly
+      env:
+        CARGO_TERM_COLOR: always
+        RUSTFLAGS: -Ctarget-feature=-crt-static
+    steps:
+      - name: install runtime
+        run: apt update && apt install -y gcc-x86-64-linux-gnu g++-x86-64-linux-gnu libc6-dev libssl-dev
+      - name: set default nightly
+        run: rustup default nightly
+      - name: smoke test
+        run: rustc --version
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: compile amd64
+        run: cargo build --target=x86_64-unknown-linux-gnu --release -p lldap -p migration-tool
+      - name: check path
+        run: ls -al target/x86_64-unknown-linux-gnu/release/
+      - name: upload amd64 lldap artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: amd64-lldap-bin
+          path: target/x86_64-unknown-linux-gnu/release/lldap
+      - name: upload amd64 migration-tool artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: amd64-migration-tool-bin
+          path: target/x86_64-unknown-linux-gnu/release/migration-tool
+
+
+  build-docker-image:
+    needs: [build-ui,build-armhf,build-aarch64,build-amd64]
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: fetch repo
+        uses: actions/checkout@v2
+        
+      - name: Download armhf lldap artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: armhf-lldap-bin
+          path: bin/armhf-bin
+      - name: Download armhf migration-tool artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: armhf-migration-tool-bin
+          path: bin/armhf-bin
+          
+      - name: Download aarch64 lldap artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: aarch64-lldap-bin
+          path: bin/aarch64-bin
+      - name: Download aarch64 migration-tool artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: aarch64-migration-tool-bin
+          path: bin/aarch64-bin   
+          
+      - name: Download amd64 lldap artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: amd64-lldap-bin
+          path: bin/amd64-bin
+      - name: Download amd64 migration-tool artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: amd64-migration-tool-bin
+          path: bin/amd64-bin
+          
+      - name: check bin path
+        run: ls -al bin/
+        
+      - name: Download llap ui artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ui
+          path: web
+      - name: setup qemu
+        uses: docker/setup-qemu-action@v1
+      - uses: docker/setup-buildx-action@v2
+      - uses: docker/metadata-action@v3
```

Comment:
The login and push should be guarded to only happen on main.

Reply:
> Nice work! I thought about making a github workflow for that, but since it's painful to test and there wasn't that much demand at the time, I didn't. Thanks for doing it :)
> 
> Quick question: how does it work with the labeling for docker images? Also, it seems like you need to guard the uploads with "only on master", see https://github.com/nitnelave/lldap/blob/main/.github/workflows/docker.yml#L42

The labeling, i usually use actions secrets like
`${{ secrets.DOCKERHUB_USERNAME }}/${{ secrets.DOCKERHUB_IMAGE_NAME }}`
But tecnically you can just override to your name `nitnelave/lldap`

I prevent to push on `docker/build-push-action`
`
          push: ${{ github.event_name != 'pull_request' }}
`
As your is preventing the login.

And `docker.yml` will obsolete, but just modify it to manual dispatch, backup is always welcome. As rust itself still pretty unstable we dont know when it breaks.

Reply:
Right, I meant tagging. Read through your PR and suggested a couple of missing tags.

If the login doesn't work, the action fails, so that's unnecessary noise in PRs, can you add a guard to the login?

Reply:
Sure 👍🏼